### PR TITLE
Introduce rake and Centroid

### PR DESCRIPTION
**Pull request by gregoryjscott**
This refactors the repository to use `rake` to drive all automation. Centroid and [config.json](https://github.com/ResourceDataInc/please/blob/rake/config.json) keep the Please specific information out of the Rakefile.

This adds necessary files/scripts to make please a NuGet package.

This also removes the NuGet packages (except for NuGet.CommandLine) from the repo. Packages used by the solution will be automatically downloaded by Visual Studio (or even the latest version of Xamarin Studio). The other option is to run `rake install`. `rake install` is required to run other rake tasks.

The rake tasks are shown below.

```
$ rake -T
rake build:debug    # Build for debugging
rake build:release  # Build for release
rake clean:all      # Clean all output
rake clean:release  # Clean release output
rake clean:test     # Clean test output
rake default        # Run tests
rake install        # Install NuGet dependencies
rake release        # Pack and push NuGet package
rake release:pack   # Pack NuGet package
rake release:push   # Push NuGet package
rake test           # Run tests
```

Tasks
- [x] ~~Document details of how everything works in a CONTRIBUTING.md file.~~ This will be handled in a separate PR.

### CODE REVIEW

**Comment by thorncp** [comment]
The description doesn't say anything about the deleted files. It looks like it's making the switch to pull dependencies from NuGet, instead of packaging them in the repo. Is that the case?

**Comment by gregoryjscott** [comment]
> The description doesn't say anything about the deleted files. It looks like it's making the switch to pull dependencies from NuGet, instead of packaging them in the repo. Is that the case?

Yes. I fixed the description.

**Comment by thorncp** [comment]
:+1:

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+  desc "Clean test output"
+  task :test do
+    clean(Config.test.output)
+  end
+
+  desc "Clean release output"
+  task :release do
+    clean(Config.release.output.prep)
+    clean(release_lib);
```
Comment (thorncp) [inline Rakefile]: What is this `;` doing here?

File: Rakefile [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    Config.release.files.each do |file|
+      FileUtils.cp file, release_lib
+    end
+    nuget("pack #{release_nuspec} -OutputDirectory #{Config.release.output.pack}")
+  end
+
+  desc "Push NuGet package"
+  task :push do
+    puts "Pushing #{release_nupkg}"
+    # nuget("push #{release_nupkg}")
```
Comment (thorncp) [inline Rakefile]: Was this commented out for testing, and then forgotten about?

File: Rakefile (hunk 1 above)
Comment (gregoryjscott) [inline Rakefile]: Stupid pinky finger...

**Pull request merged**